src/lsdb/catalog/catalog.py

```diff
@@ -248,6 +251,31 @@ def cone_search(self, ra: float, dec: float, radius: float):
         ddf_partition_map = {pixel: i for i, pixel in enumerate(pixels_in_cone)}
         return Catalog(cone_search_ddf, ddf_partition_map, filtered_hc_structure)
 
+    def polygon_search(self, polygon: SingleSphericalPolygon) -> Catalog:
```

Comment:
Same comment about the polygon input types as with hipscat, could you make an issue for it here too?

Reply:
Created the issue with a pointer to the one on hipscat (#81).